File: go/raindrops/raindrops.go
// Package raindrops provides Convert function.
package raindrops

import "strconv"

// Convert function converts numbers into a string.
func Convert(num int) string {
	var result string

	if num%3 == 0 {
		result += "Pling"
	}
	if num%5 == 0 {
		result += "Plang"
	}
	if num%7 == 0 {
		result += "Plong"
	}
	if result == "" {
		result += strconv.Itoa(num)
	}
	return result
}

File: ruby/raindrops/raindrops.rb
class Raindrops
  class << self
    FACTOR_to_SOUND = {
      3 => 'Pling',
      5 => 'Plang',
      7 => 'Plong'
    }.freeze

    def convert(number)
      result = ''
      result << FACTOR_to_SOUND[3] if (number % 3).zero?
      result << FACTOR_to_SOUND[5] if (number % 5).zero?
      result << FACTOR_to_SOUND[7] if (number % 7).zero?
      result << number.to_s if result.empty?
      result
    end
  end
end
